pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -1648,6 +1648,15 @@ private void makeStreamConsumer(String reason) {
     _segmentLogger.info("Creating new stream consumer for topic partition {} , reason: {}", _clientId, reason);
     _partitionGroupConsumer =
         _streamConsumerFactory.createPartitionGroupConsumer(_clientId, _partitionGroupConsumptionStatus);
+    try {
+      _partitionGroupConsumer.validateStreamState(_currentOffset);
+      _serverMetrics.setValueOfPartitionGauge(_tableNameWithType, 0,
```

Comment:
So it's good that we have a gauge per table (because otherwise, a table which is healthy could reset the gauge from `1` to `0` if it executes this check _after_ an unhealthy table set the gauge to `1`).

However, what happens if there are multiple partitions on a table and one partition's segment sets this gauge to `1`, then a different partition's segment executes this code and resets the gauge from `1` to `0`? Put another way: if a table has 3 partitions and 1 of them fails `validateStreamState` but the other two pass `validateStreamState` then it's very likely that the healthy partitions will mask the unhealthy partition.

We need to make sure that if there is a segment that fails this validation this gauge is set to `1` until that segment is fixed or there is some other manual intervention.

Reply:
The gauge for a specific partition. `setValueOfParitionGauge` takes a partition id as one of the parameters.

Reply:
Going back on this as well. Metrics/alerts per partition is not allowed anymore. So the metric/alert is for a table. Logs have more information.